# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: tools/profiler/gecko/nsProfiler.cpp
# Commit: ac548c2a725b
# Full Hash: ac548c2a725b66db57c81bb24accca2bbe075931
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2019-07-29 21:38:39
# Description:
#   Bug 1565930, try to make Promise less error prone to compartment mismatches, r=bzbarsky
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38762
# ==============================================================================

diff -r c4980c083852 -r ac548c2a725b tools/profiler/gecko/nsProfiler.cpp
--- a/tools/profiler/gecko/nsProfiler.cpp	Mon Jul 29 08:00:35 2019 +0000
+++ b/tools/profiler/gecko/nsProfiler.cpp	Mon Jul 29 14:43:54 2019 +0000
@@ -308,7 +308,7 @@
                   MOZ_ASSERT(gotException);
 
                   jsapi.ClearException();
-                  promise->MaybeReject(cx, exn);
+                  promise->MaybeReject(exn);
                 }
               } else {
                 promise->MaybeResolve(val);